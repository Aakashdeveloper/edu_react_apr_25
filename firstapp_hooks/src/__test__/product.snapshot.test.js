import React from 'react';
import ProductDisplay from '../component/ProductDisplay';
import {render} from '@testing-library/react';

describe('Product Display Snapshot',() => {

    let mockData = [
        {
            "id":1,
            "brand":"3M",
            "discount":"Less than 10%",
            "form": "Mask",
            "uses":"Respiratory Care",
            "name":"3M 9004IN Particle Respirator Mask",
            "image":"https://i.ibb.co/8BJhgK2/r6xkegcloxgn5eq8owwg.jpg",
            "description":"box of 10 masks",
            "cost":180
        },
        {
            "id":2,
            "brand":"Vogmask",
            "discount":"10% and above",
            "form": "Mask",
            "uses":"Respiratory Care",
            "name":"Vogmask Vector Havoc N99CV Mask Large",
            "image":"https://i.ibb.co/fHQ8JY4/kclatp7d9dt0dwgu4zkq.jpg",
            "description":"box of 10 masks",
            "cost":200
        }
    ]
    test('Testing Footer',() => {
        const {asFragment} = render(<ProductDisplay products={mockData}/>)
        expect(asFragment()).toMatchSnapshot()
    })
})
