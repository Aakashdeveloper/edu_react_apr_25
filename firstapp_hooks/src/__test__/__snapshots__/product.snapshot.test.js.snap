// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Display Snapshot Testing Footer 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="card"
    >
      <img
        alt="3M 9004IN Particle Respirator Mask"
        src="https://i.ibb.co/8BJhgK2/r6xkegcloxgn5eq8owwg.jpg"
      />
      <div>
        <h3>
          3M 9004IN Particle Respirator Mask
        </h3>
        <h3>
          box of 10 masks
        </h3>
        <h3>
          Rs. 180
        </h3>
        <h3>
          Respiratory Care
        </h3>
      </div>
    </div>
    <div
      class="card"
    >
      <img
        alt="Vogmask Vector Havoc N99CV Mask Large"
        src="https://i.ibb.co/fHQ8JY4/kclatp7d9dt0dwgu4zkq.jpg"
      />
      <div>
        <h3>
          Vogmask Vector Havoc N99CV Mask Large
        </h3>
        <h3>
          box of 10 masks
        </h3>
        <h3>
          Rs. 200
        </h3>
        <h3>
          Respiratory Care
        </h3>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
